[SPARK-26797][SQL][WIP][test-maven] Start using the new logical types API of Parquet 1.11.0 instead of the deprecated one

nandorKollar opened the pull request:
## What changes were proposed in this pull request?

A new, more flexible logical type API was introduced in parquet-mr 1.11.0 (based on the the Thrift field in parquet-format available for a while). This change migrates from the old (now deprecated) enum-based OriginalType API to this new logical type API.

In addition to replacing the deprecated API calls, this PR also introduces support for reading the new subtypes for different timestamp semantics.

Since parquet-mr 1.11.0 is not yet released, this is tested against a release candidate. Before merging, the additional repository should be deleted from pom.xml, which can only be done once parquet-mr 1.11.0 is released.

## How was this patch tested?

Unit tests were added to the PR.

nkollar committed:
Timezone adjustment

nkollar committed:
Generate file with epoch (called rawValue) value as id

nkollar committed:
Fix predicate pushdown - equals

nkollar committed:
refactor

nkollar committed:
Parquet pushdown - without lt and lteq predicates

nkollar committed:
Fix lt and lteq predicates with timestamps

nkollar committed:
Fix style issues, remove whitespaces

nkollar committed:
Fix failing tests

nkollar committed:
Upgrade to Parquet 1.11.0 release candidate

AmplabJenkins commented:
Can one of the admins verify this patch?

nkollar committed:
Fix name

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

attilapiros commented on sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java:
```diff
@@ -425,13 +438,31 @@ private void readLongBatch(int rowId, int num, WritableColumnVector column) thro
+        for (int i = 0; i < num; i++) {
+          if (defColumn.readInteger() == maxDefLevel) {
+            long timestamp = dataColumn.readLong();
+            if (needsTimezoneAdjustment()) {
```
Unnecessary if condition as`needsTimezoneAdjustment()` will be always true because in line 442 it is already tested.

attilapiros commented on sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala:
```diff
@@ -100,6 +104,28 @@ private[parquet] class ParquetFilters(
 
+  private def timestampValue(timestampType: TimestampLogicalTypeAnnotation, v: Any): JLong =
+    if (timestampType.getUnit == TimeUnit.MICROS) {
+      Option(v).map(
```
Here within the `map` the if computes a transformation function. I think this is a bit complex and can be simplified especially as both branches of the outer if ends with an orNull `Option(v).map(...).orNull`.

I would suggest to first check whether v is null (you can keep using the `Option` with `map`) and just latter check for the TimeUnit.

A multiline `map` expression preferred to be written in the style:
```
map { v =>
  <transformation>
}
```

nandorKollar replied:
Thanks, I also replaced the if to a match expression. With if, the code would silently convert NANO precision timestamps incorrectly, that's a bug, I think throwing an exception in this case is a better option.

attilapiros commented on sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadSupport.scala:
```diff
@@ -49,15 +50,16 @@ import org.apache.spark.sql.types._
  */
-private[parquet] class ParquetReadSupport(val convertTz: Option[TimeZone])
+private[parquet] class ParquetReadSupport(val convertTz: Option[TimeZone],
+                                          val sessionLocalTz: TimeZone)
```
nit: indentation, avoid aligning parameters into one row. 
It is preferred to have each parameters to have its own line if they do not fit in one line.

attilapiros commented on sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala:
```diff
@@ -408,7 +410,8 @@ class ParquetFileFormat
       val taskContext = Option(TaskContext.get())
       if (enableVectorizedReader) {
         val vectorizedReader = new VectorizedParquetRecordReader(
```
Nit: each parameter should be in its own line

nandorKollar commented:
Thanks @attilapiros for the review, I'll address the issues you find soon.

attilapiros commented on sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java:
```diff
@@ -441,6 +472,16 @@ private void readLongBatch(int rowId, int num, WritableColumnVector column) thro
+      !((TimestampLogicalTypeAnnotation) logicalTypeAnnotation).isAdjustedToUTC();
+  }
+
+  private boolean isTimestampWithUnit(LogicalTypeAnnotation logicalTypeAnnotation, LogicalTypeAnnotation.TimeUnit timeUnit) {
```
Nit: too long line

attilapiros commented on sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala:
```diff
@@ -257,17 +258,31 @@ private[parquet] class ParquetRowConverter(
 
-      case TimestampType if parquetType.getOriginalType == OriginalType.TIMESTAMP_MICROS =>
+      case TimestampType
+        if parquetType.getLogicalTypeAnnotation.isInstanceOf[TimestampLogicalTypeAnnotation] &&
```
I suggest to avoid these long expression by calling a new method after the if.
Its signature can be something like: 
```
  def isTimestampWithUnit(parquetType: Type, timeUnit: LogicalTypeAnnotation.TimeUnit): Boolean
```

attilapiros commented on sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala:
```diff
@@ -257,17 +258,31 @@ private[parquet] class ParquetRowConverter(
 
-      case TimestampType if parquetType.getOriginalType == OriginalType.TIMESTAMP_MILLIS =>
+      case TimestampType
+        if parquetType.getLogicalTypeAnnotation.isInstanceOf[TimestampLogicalTypeAnnotation] &&
```
Same as above.

attilapiros commented on sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala:
```diff
@@ -301,7 +316,8 @@ private[parquet] class ParquetRowConverter(
       // elements where the element type is the type of the field.
-      case t: ArrayType if parquetType.getOriginalType != LIST =>
+      case t: ArrayType
+        if !parquetType.getLogicalTypeAnnotation.isInstanceOf[ListLogicalTypeAnnotation] =>
```
Nit: the two ifs indented in the same level looks suspicious. I suggest to keep the if at the line of case and indent the condition with 4 spaces.

attilapiros replied:
Like:
```
 case t: ArrayType if 
     !parquetType.getLogicalTypeAnnotation.isInstanceOf[ListLogicalTypeAnnotation] =>
```

attilapiros commented on sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala:
```diff
@@ -316,7 +332,8 @@ private[parquet] class ParquetRowConverter(
       case t: StructType =>
         new ParquetRowConverter(
-          schemaConverter, parquetType.asGroupType(), t, convertTz, new ParentContainerUpdater {
+          schemaConverter, parquetType.asGroupType(), t, convertTz, sessionLocalTz,
```
Every params should be in its own line.

attilapiros commented on sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala:
```diff
@@ -42,7 +44,8 @@ class ParquetEncodingSuite extends ParquetCompatibilityTest with SharedSQLContex
         val conf = sqlContext.conf
         val reader = new VectorizedParquetRecordReader(
-          null, conf.offHeapColumnVectorEnabled, conf.parquetVectorizedReaderBatchSize)
+          null, TimeZone.getDefault, conf.offHeapColumnVectorEnabled,
```
Every params should be in its own line.

attilapiros commented on sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala:
```diff
@@ -69,7 +72,8 @@ class ParquetEncodingSuite extends ParquetCompatibilityTest with SharedSQLContex
         val conf = sqlContext.conf
         val reader = new VectorizedParquetRecordReader(
-          null, conf.offHeapColumnVectorEnabled, conf.parquetVectorizedReaderBatchSize)
+          null, TimeZone.getDefault, conf.offHeapColumnVectorEnabled,
```
Same as above.

attilapiros commented on sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala:
```diff
@@ -100,7 +104,8 @@ class ParquetEncodingSuite extends ParquetCompatibilityTest with SharedSQLContex
         val conf = sqlContext.conf
         val reader = new VectorizedParquetRecordReader(
-          null, conf.offHeapColumnVectorEnabled, conf.parquetVectorizedReaderBatchSize)
+          null, TimeZone.getDefault, conf.offHeapColumnVectorEnabled,
```
Same as above.

nkollar committed:
Address code review comments

nandorKollar commented:
Thanks @attilapiros for the review, fixed your findings.

squito commented:
CAn you mention briefly how backwards compatibility will work after these changes?  I assume this is somehow handled within parquet itself -- just a pointer to the relevant info would help.

Also I assume I should not bother triggering tests yet, as automated builds will fail without a published version of parquet?

nandorKollar commented:
@dongjoon-hyun @squito could you please review this PR?

nandorKollar commented:
@squito parquet-mr 1.11.0 writes both the old and the new logical types (converted_type and logicalType) in the Thrift schema, so old readers (who know only about converted_type) are able to read the annotation as long as there's a corresponding logicalType for the converted_type. Parquet-mr handles this conversion internally. For all legacy converted_type there's a corresponding logicalType, but since converted_type are deprecated, newly introduce logicalTypes might not have corresponding converted_type (for example timestamp with nano precision doesn't have any corresponding converted_type). In this case old readers will just see the physical type.

As of reading old files where new logical types are not present in the schema, only converted_type is taken into account, and parquet-mr takes care of the conversion to logical type representation internally. The conversion rules between original_types and logicalTypes are documented in [parquet-format](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md). Does this answer your question?

dongjoon-hyun commented:
ok to test.

dongjoon-hyun commented:
Thank you for pinging me, @nandorKollar . Let's ping @rdblue since this is Parquet 1.11.0.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/7752/
Test PASSed.

SparkQA commented:
**[Test build #102094 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102094/testReport)** for PR 23721 at commit [`b9ced55`](https://github.com/apache/spark/commit/b9ced55b2cf39187ffc6660835bb8e679fbd9cfd).

SparkQA commented:
**[Test build #102094 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102094/testReport)** for PR 23721 at commit [`b9ced55`](https://github.com/apache/spark/commit/b9ced55b2cf39187ffc6660835bb8e679fbd9cfd).
 * This patch **fails build dependency tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `sealed trait MLEvent extends SparkListenerEvent `
  * `case class TransformStart() extends MLEvent `
  * `case class TransformEnd() extends MLEvent `
  * `case class FitStart[M <: Model[M]]() extends MLEvent `
  * `case class FitEnd[M <: Model[M]]() extends MLEvent `
  * `case class LoadInstanceStart[T](path: String) extends MLEvent `
  * `case class LoadInstanceEnd[T]() extends MLEvent `
  * `case class SaveInstanceStart(path: String) extends MLEvent `
  * `case class SaveInstanceEnd(path: String) extends MLEvent `
  * `class _DynamicModuleFuncGlobals(dict):`
  * `class LogisticRegressionModel(JavaModel, JavaClassificationModel, JavaMLWritable, JavaMLReadable,`
  * `class GaussianMixtureModel(JavaModel, JavaMLWritable, JavaMLReadable, HasTrainingSummary):`
  * `class KMeansModel(JavaModel, GeneralJavaMLWritable, JavaMLReadable, HasTrainingSummary):`
  * `class BisectingKMeansModel(JavaModel, JavaMLWritable, JavaMLReadable, HasTrainingSummary):`
  * `class LinearRegressionModel(JavaModel, JavaPredictionModel, GeneralJavaMLWritable, JavaMLReadable,`
  * `class HasTrainingSummary(object):`

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102094/
Test FAILed.

attilapiros commented:
@nandorKollar I would try to rebase this PR on top the master (as you have not touched the above classes).

So update your fork with the official spark then execute this on your branch: 
```
$ git rebase origin/master
```

nkollar committed:
Add staging repository to use parquet-mr 1.11.0 RC3

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/7755/
Test PASSed.

SparkQA commented:
**[Test build #102097 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102097/testReport)** for PR 23721 at commit [`4d8fc37`](https://github.com/apache/spark/commit/4d8fc3788200722446064119dce5a2c8d8847c0c).

SparkQA commented:
**[Test build #102097 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102097/testReport)** for PR 23721 at commit [`4d8fc37`](https://github.com/apache/spark/commit/4d8fc3788200722446064119dce5a2c8d8847c0c).
 * This patch **fails build dependency tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102097/
Test FAILed.

nkollar committed:
Update manifest

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/7756/
Test PASSed.

SparkQA commented:
**[Test build #102098 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102098/testReport)** for PR 23721 at commit [`5a3df02`](https://github.com/apache/spark/commit/5a3df029d0111294126d87f6c915637aa8944e20).

SparkQA commented:
**[Test build #102098 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102098/testReport)** for PR 23721 at commit [`5a3df02`](https://github.com/apache/spark/commit/5a3df029d0111294126d87f6c915637aa8944e20).
 * This patch **fails to build**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102098/
Test FAILed.

nandorKollar commented:
Retest this please.

attilapiros commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/7758/
Test PASSed.

SparkQA commented:
**[Test build #102102 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102102/testReport)** for PR 23721 at commit [`5a3df02`](https://github.com/apache/spark/commit/5a3df029d0111294126d87f6c915637aa8944e20).

SparkQA commented:
**[Test build #102102 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102102/testReport)** for PR 23721 at commit [`5a3df02`](https://github.com/apache/spark/commit/5a3df029d0111294126d87f6c915637aa8944e20).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102102/
Test FAILed.

rdblue commented:
Thanks for pinging me, @dongjoon-hyun. I'd definitely like to review this but probably not before 1.11.0 is released. Until then, I think time is better spent validating the release.

dongjoon-hyun commented on dev/deps/spark-deps-hadoop-2.6:
```diff
@@ -0,0 +1,198 @@
+JavaEWAH-0.3.2.jar
```
Hi, @nandorKollar . Please remove this file because this is for Spark 3.0.

dongjoon-hyun commented on dev/deps/spark-deps-hadoop-2.7:
```diff
@@ -17,6 +17,7 @@ arpack_combined_all-0.1.jar
 arrow-format-0.12.0.jar
 arrow-memory-0.12.0.jar
 arrow-vector-0.12.0.jar
+audience-annotations-0.7.0.jar
```
Is it required at Spark? Otherwise, can we exclude this in Parquet 1.11.0 RC?

nandorKollar replied:
@dongjoon-hyun it is a new dependency for Parquet to annotate which API is public and what is private. I don't think that we can exclude it from Parquet, since it is used there. The annotations are Yetus audience annotations with runtime retention policy (I think class retention policy would be more adequate), therefore I'm afraid the had to be present at runtime.

squito commented:
I'm not sure why jenkins doesn't give a test report, but there are real failures in that test run:

* InMemoryColumnarQuerySuite "SPARK-22673: InMemoryRelation should utilize existing stats of the plan to be cached"
* ParquetInteroperabilitySuite "parquet timestamp read path"
* HadoopFsRelationSuite "SPARK-22790: spark.sql.sources.compressionFactor takes effect"
* ParquetPartitionDiscoverySuite "column type inference" & "parse partitions"
* HDFSMetadataLogSuite "HDFSMetadataLog: metadata directory collision"
* ParquetQuerySuite "SPARK-26677: negated null-safe equality comparison should not filter matched row groups"

dongjoon-hyun commented:
@nandorKollar . Before updating this PR, please test and pass the above test suites (mentioned by @squito) in your local environment first.

nkollar committed:
Fix failing tests, address code review comments

SparkQA commented:
**[Test build #102256 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102256/testReport)** for PR 23721 at commit [`b78cb04`](https://github.com/apache/spark/commit/b78cb04721c3f3237da69ae784f29d58790a0103).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/7863/
Test PASSed.

nandorKollar commented:
@dongjoon-hyun @squito triaged the failures:
* InMemoryColumnarQuerySuite - failed because of change in Parquet file size due to column indexes
* ParquetInteroperabilitySuite "parquet timestamp read path" - _had to regenerate tests files, those which are attached to the PR were generated before RC_
* HadoopFsRelationSuite "SPARK-22790: spark.sql.sources.compressionFactor takes effect" - _failed because of change in Parquet file size, due to column indexes_
* ParquetPartitionDiscoverySuite "column type inference" & "parse partitions" - _couldn't reproduce test locally, it passes on my machine_
* HDFSMetadataLogSuite "HDFSMetadataLog: metadata directory collision" - _couldn't reproduce, in Jenkins logs I could see "The await method on Waiter timed out.", maybe it is a flaky test? Looks like an unrelated failure for me._
* ParquetQuerySuite "SPARK-26677: negated null-safe equality comparison should not filter matched row groups" - _fails because PARQUET-1510 is not yet included into 1.11.0 RC3, RC4 is needed (this bug was found after RC3 was created)_

SparkQA commented:
**[Test build #102256 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102256/testReport)** for PR 23721 at commit [`b78cb04`](https://github.com/apache/spark/commit/b78cb04721c3f3237da69ae784f29d58790a0103).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102256/
Test FAILed.

nkollar committed:
Fix failing ParquetPartitionDiscoverySuite test

ParquetPartitionDiscoverySuite failed when executed after ParquetInteroperabilitySuite using Maven
The reasion for that is, that ParquetInteroperabilitySuite changes the timezone in one test case,
but doesn't restore the original. This could be easily fixed by restoring the original timezone in
a finally block.

SparkQA commented:
**[Test build #102478 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102478/testReport)** for PR 23721 at commit [`7866e62`](https://github.com/apache/spark/commit/7866e62aa7b396b73314b14d24fce08bebfb2d02).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8051/
Test PASSed.

SparkQA commented:
**[Test build #102478 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102478/testReport)** for PR 23721 at commit [`7866e62`](https://github.com/apache/spark/commit/7866e62aa7b396b73314b14d24fce08bebfb2d02).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102478/
Test FAILed.

nkollar committed:
Use direct snapshot url

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8114/
Test PASSed.

SparkQA commented:
**[Test build #102553 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102553/testReport)** for PR 23721 at commit [`2a58bb8`](https://github.com/apache/spark/commit/2a58bb8bc4bbce229d558107f4ed43b4d94b6d0d).

SparkQA commented:
**[Test build #102553 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102553/testReport)** for PR 23721 at commit [`2a58bb8`](https://github.com/apache/spark/commit/2a58bb8bc4bbce229d558107f4ed43b4d94b6d0d).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102553/
Test FAILed.

squito commented:
I'm still not sure why jenkins is not displaying the test failures normally, but ParquetQuerySuites "SPARK-26677: negated null-safe equality comparison should not filter matched row groups" is still failing

nandorKollar commented:
@squito I'm a bit worried about this failure, because in the latest Parquet RC this should be fine. Either there's some Maven cache which is not cleaned, an older Parquet RC is stuck there, or the fix is not in the release candidate. I expected a green build now.

squito commented:
The pom still says `<name>Parquet 1.11.0 RC3</name>` -- does that matter?  do you need to do something else to pick up RC4?

nandorKollar commented:
@squito I don't think it matters. I had an earlier Parquet RC in my .m2 cache, the tests failed, but when I manually deleted the cached version, the problematic test passed. I think executing mvn dependency:purge-local-repository on the Jenkins node would solve this problem.

nkollar committed:
Change updatePolicy to always

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8344/
Test PASSed.

SparkQA commented:
**[Test build #102856 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102856/testReport)** for PR 23721 at commit [`e121b55`](https://github.com/apache/spark/commit/e121b552eaff9bc42f155e5a2a2a4ea48e3439d9).

SparkQA commented:
**[Test build #102856 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102856/testReport)** for PR 23721 at commit [`e121b55`](https://github.com/apache/spark/commit/e121b552eaff9bc42f155e5a2a2a4ea48e3439d9).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102856/
Test FAILed.

attilapiros commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8347/
Test PASSed.

SparkQA commented:
**[Test build #102860 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102860/testReport)** for PR 23721 at commit [`e121b55`](https://github.com/apache/spark/commit/e121b552eaff9bc42f155e5a2a2a4ea48e3439d9).

SparkQA commented:
**[Test build #102860 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102860/testReport)** for PR 23721 at commit [`e121b55`](https://github.com/apache/spark/commit/e121b552eaff9bc42f155e5a2a2a4ea48e3439d9).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102860/
Test FAILed.

attilapiros commented:
This Py4JNetworkError must be unrelated:

```
Traceback (most recent call last):
  File "/home/jenkins/workspace/SparkPullRequestBuilder/python/lib/py4j-0.10.8.1-src.zip/py4j/java_gateway.py", line 1014, in send_command
    response = connection.send_command(command)
  File "/home/jenkins/workspace/SparkPullRequestBuilder/python/lib/py4j-0.10.8.1-src.zip/py4j/java_gateway.py", line 1193, in send_command
    "Error while receiving", e, proto.ERROR_ON_RECEIVE)
Py4JNetworkError: Error while receiving
```

attilapiros commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8360/
Test PASSed.

SparkQA commented:
**[Test build #102872 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102872/testReport)** for PR 23721 at commit [`e121b55`](https://github.com/apache/spark/commit/e121b552eaff9bc42f155e5a2a2a4ea48e3439d9).

SparkQA commented:
**[Test build #102872 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102872/testReport)** for PR 23721 at commit [`e121b55`](https://github.com/apache/spark/commit/e121b552eaff9bc42f155e5a2a2a4ea48e3439d9).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102872/
Test FAILed.

attilapiros commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8385/
Test PASSed.

SparkQA commented:
**[Test build #102910 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102910/testReport)** for PR 23721 at commit [`e121b55`](https://github.com/apache/spark/commit/e121b552eaff9bc42f155e5a2a2a4ea48e3439d9).

SparkQA commented:
**[Test build #102910 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102910/testReport)** for PR 23721 at commit [`e121b55`](https://github.com/apache/spark/commit/e121b552eaff9bc42f155e5a2a2a4ea48e3439d9).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102910/
Test FAILed.

nandorKollar commented:
retest this please

SparkQA commented:
**[Test build #102988 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102988/testReport)** for PR 23721 at commit [`1880d6c`](https://github.com/apache/spark/commit/1880d6c48f6440eca12c452b28d1ea0cb4b9ffde).

SparkQA commented:
**[Test build #102988 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102988/testReport)** for PR 23721 at commit [`1880d6c`](https://github.com/apache/spark/commit/1880d6c48f6440eca12c452b28d1ea0cb4b9ffde).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102988/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8456/
Test PASSed.

nkollar committed:
remove duplicated repository

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8459/
Test PASSed.

SparkQA commented:
**[Test build #102991 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102991/testReport)** for PR 23721 at commit [`b7e1915`](https://github.com/apache/spark/commit/b7e19151d4a033045b6fe6388ccc67b2d4a10f8a).

SparkQA commented:
**[Test build #102991 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/102991/testReport)** for PR 23721 at commit [`b7e1915`](https://github.com/apache/spark/commit/b7e19151d4a033045b6fe6388ccc67b2d4a10f8a).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/102991/
Test FAILed.

nkollar committed:
Use 1.12.0-SNAPSHOT

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8518/
Test PASSed.

SparkQA commented:
**[Test build #103056 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103056/testReport)** for PR 23721 at commit [`5f0393e`](https://github.com/apache/spark/commit/5f0393e7dd1fe4734522c71d94e7df7c891b7133).

SparkQA commented:
**[Test build #103056 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103056/testReport)** for PR 23721 at commit [`5f0393e`](https://github.com/apache/spark/commit/5f0393e7dd1fe4734522c71d94e7df7c891b7133).
 * This patch **fails build dependency tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/103056/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8589/
Test PASSed.

SparkQA commented:
**[Test build #103132 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103132/testReport)** for PR 23721 at commit [`dc22471`](https://github.com/apache/spark/commit/dc224714664ca76d101755b97828ebdb6232667e).

nkollar committed:
update deps file

SparkQA commented:
**[Test build #103135 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103135/testReport)** for PR 23721 at commit [`e0fbf44`](https://github.com/apache/spark/commit/e0fbf441efe8e52a6b167e033aaad71207d75001).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/8592/
Test PASSed.

SparkQA commented:
**[Test build #103132 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103132/testReport)** for PR 23721 at commit [`dc22471`](https://github.com/apache/spark/commit/dc224714664ca76d101755b97828ebdb6232667e).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/103132/
Test FAILed.

SparkQA commented:
**[Test build #103135 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103135/testReport)** for PR 23721 at commit [`e0fbf44`](https://github.com/apache/spark/commit/e0fbf441efe8e52a6b167e033aaad71207d75001).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/103135/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/9268/
Test PASSed.

SparkQA commented:
**[Test build #103905 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103905/testReport)** for PR 23721 at commit [`6e95803`](https://github.com/apache/spark/commit/6e95803ae9cc18c2b34bb7992357dca35755206e).

SparkQA commented:
**[Test build #103905 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103905/testReport)** for PR 23721 at commit [`6e95803`](https://github.com/apache/spark/commit/6e95803ae9cc18c2b34bb7992357dca35755206e).
 * This patch **fails build dependency tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/103905/
Test FAILed.

nkollar committed:
fix manifest file

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/9270/
Test PASSed.

SparkQA commented:
**[Test build #103907 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103907/testReport)** for PR 23721 at commit [`9a39876`](https://github.com/apache/spark/commit/9a39876fa88c6311ee52f8453c4ed0bbc18d553a).

SparkQA commented:
**[Test build #103907 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103907/testReport)** for PR 23721 at commit [`9a39876`](https://github.com/apache/spark/commit/9a39876fa88c6311ee52f8453c4ed0bbc18d553a).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/103907/
Test FAILed.

nkollar committed:
purge Parquet artifacts from local repository

SparkQA commented:
**[Test build #103917 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103917/testReport)** for PR 23721 at commit [`ca9bdc3`](https://github.com/apache/spark/commit/ca9bdc31c9fcbb35b7c619f176916096f1c822bc).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/103917/
Test FAILed.

nkollar committed:
line too long

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/9278/
Test FAILed.

SparkQA commented:
**[Test build #103918 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103918/testReport)** for PR 23721 at commit [`f8ecac1`](https://github.com/apache/spark/commit/f8ecac15542d15805ea733781b501b4e6645e892).

SparkQA commented:
**[Test build #103918 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103918/testReport)** for PR 23721 at commit [`f8ecac1`](https://github.com/apache/spark/commit/f8ecac15542d15805ea733781b501b4e6645e892).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/103918/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/9279/
Test PASSed.

squito commented:
Sorry @nandorKollar, real python style failures.  you can run `dev/lint-python` locally

```
pycodestyle checks failed:
./dev/run-tests.py:298:1: E101 indentation contains mixed spaces and tabs
./dev/run-tests.py:298:1: W191 indentation contains tabs
./dev/run-tests.py:298:2: E128 continuation line under-indented for visual indent
./dev/run-tests.py:299:1: E101 indentation contains mixed spaces and tabs
```

nkollar committed:
fix python style error

nandorKollar commented:
Thanks @squito hope I fixed every Python error.

SparkQA commented:
**[Test build #103919 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103919/testReport)** for PR 23721 at commit [`c9a50e4`](https://github.com/apache/spark/commit/c9a50e4b21cb5e30fcfe3d9415625735f648f20b).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/9280/
Test PASSed.

SparkQA commented:
**[Test build #103919 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/103919/testReport)** for PR 23721 at commit [`c9a50e4`](https://github.com/apache/spark/commit/c9a50e4b21cb5e30fcfe3d9415625735f648f20b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/103919/
Test PASSed.

nandorKollar commented:
Tests passed with parquet-mr 1.11.0 release candidate, pinging @squito @rdblue @dongjoon-hyun, @HyukjinKwon for review.

felixcheung commented:
that's great, honestly we can't merge this until parquet 1.11 is released officially.

nandorKollar commented:
@felixcheung yes, I know, that's why it is still tagged with WIP. I opened the PR before the official release, because I thought that until it gets released I could get some useful feedback.

HyukjinKwon reviewed (commented):
Not a big deal but `sql/core/src/test/resources/test-data/timestamp_dictionary.parq` - should it be `timestamp_dictionary.parquet`?

HyukjinKwon commented:
cc @cloud-fan, @wangyum, and @liancheng FYI

gatorsmile commented:
Anybody is using Parquet 1.11 in their production systems? What is the major motivation to upgrade the parquet version just after a new version of Parquet is released?

zivanfi commented:
Parquet 1.11 is not released yet, but the release candidate provides [several new features, improvements and bugfixes](https://github.com/apache/parquet-mr/blob/master/CHANGES.md), most notably:

- A new (but backwards- and forwards-compatible) logical type system that allows representation of timestamps with different semantics.
- Column indexes, which allows pinpointing pages matching query conditions, leading to significant performance improvements in highly selective queries.

nandorKollar commented:
I recently found an other reason for upgrading Parquet version to 1.11.0 (once released): due to PARQUET-1472 when reading decimals with fixed size byte array underlying physical type, the dictionary filter could incorrectly drop row groups, silently giving back wrong results. The fix for this problem is only present in 1.11.0 as of now.

felixcheung commented:
what's the status of parquet 1.11?

AmplabJenkins commented:
Can one of the admins verify this patch?

wangyum commented:
@felixcheung It's still in voting: http://mail-archives.apache.org/mod_mbox/parquet-dev/201911.mbox/%3CCAAhnZP7Y-%2BY_Y-jagAx0G5TEczyfDAdAQt7QdBbk-7pEjiPbXg%40mail.gmail.com%3E

wangyum commented:
retest this please

SparkQA commented:
**[Test build #113713 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/113713/testReport)** for PR 23721 at commit [`c9a50e4`](https://github.com/apache/spark/commit/c9a50e4b21cb5e30fcfe3d9415625735f648f20b).

AmplabJenkins commented:
Build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/18593/
Test PASSed.

SparkQA commented:
**[Test build #113713 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/113713/testReport)** for PR 23721 at commit [`c9a50e4`](https://github.com/apache/spark/commit/c9a50e4b21cb5e30fcfe3d9415625735f648f20b).
 * This patch **fails build dependency tests**.
 * This patch **does not merge cleanly**.
 * This patch adds no public classes.

AmplabJenkins commented:
Build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/113713/
Test FAILed.

nandorKollar committed:
Remove snapshot repo

Parquet 1.11.0 is officially released, no need to use snapshot.

SparkQA commented:
**[Test build #116098 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/116098/testReport)** for PR 23721 at commit [`9dbf152`](https://github.com/apache/spark/commit/9dbf1523ed247dec11d21217241bfb6014dc39ee).

SparkQA commented:
**[Test build #116098 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/116098/testReport)** for PR 23721 at commit [`9dbf152`](https://github.com/apache/spark/commit/9dbf1523ed247dec11d21217241bfb6014dc39ee).
 * This patch **fails build dependency tests**.
 * This patch **does not merge cleanly**.
 * This patch adds no public classes.

AmplabJenkins commented:
Build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/116098/
Test FAILed.

h-vetinari commented:
What's the current status of this PR?

Parquet 1.11 has been release, but spark is still [waiting](https://issues.apache.org/jira/browse/SPARK-26346) for 1.11.1. Incidentally, the parquet 1.11.1 release is [waiting](https://issues.apache.org/jira/browse/SPARK-26346?focusedCommentId=17039871&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17039871) for spark feedback on the snapshot. Maybe that's could be done here?

h-vetinari commented:
@nandorKollar 
Should this PR be revived? Would you mind if I open a new one based on your changes? I tried rebasing your branch on master, but there's a whole lot of conflicts, so before I try to resolve them, I wanted to ask what the status is here.

CC @dongjoon-hyun @wangyum

sunchao commented:
+1. Parquet community (format/mr/cpp etc) has moved from the original enum-based converted type to the new union-based logical type with richer metadata, so it'd be great to see Spark adapt that too.

h-vetinari commented:
Attempt to carry this PR in #31685.
